.azure-pipelines/sign-binary.yaml

```diff
@@ -0,0 +1,225 @@
+          AnalyzeVerbose: true
+          AnalyzeHashes: true
+          AnalyzeEnvironment: false
+      - task: PublishSecurityAnalysisLogs@3
+        displayName: 'Publish Security Analysis Logs'
+        inputs:
+          ArtifactName: SecurityLogs
+
+      - task: PostAnalysis@1
+        displayName: 'Post Analysis'
+        inputs:
+          BinSkim: true
+      - powershell: echo "##vso[task.setvariable variable=cliArchiveName;isOutput=true]$(cliArchiveNameVar)"
+        name: setArchiveNameStep
+
+- stage: sign
+  dependsOn: download
+  jobs:
+    - job: esrpSign
+      steps:
+      - checkout: none
+      - download: current
+        artifact: unzipped
+      - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
+        displayName: 'ESRP CodeSigning'
+        inputs:
+          ConnectedServiceName: 'microsoftgraph ESRP CodeSign DLL and NuGet (AKV)'
+          FolderPath: '$(Pipeline.Workspace)\unzipped'
+          signConfigType: inlineSignParams
+          inlineOperation: |
+            [
+                {
+                    "keyCode": "CP-230012",
+                    "operationSetCode": "SigntoolSign",
+                    "parameters": [
+                    {
+                        "parameterName": "OpusName",
+                        "parameterValue": "Microsoft"
+                    },
+                    {
+                        "parameterName": "OpusInfo",
+                        "parameterValue": "http://www.microsoft.com"
+                    },
+                    {
+                        "parameterName": "FileDigest",
+                        "parameterValue": "/fd \"SHA256\""
+                    },
+                    {
+                        "parameterName": "PageHash",
+                        "parameterValue": "/NPH"
+                    },
+                    {
+                        "parameterName": "TimeStamp",
+                        "parameterValue": "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+                    }
+                    ],
+                    "toolName": "sign",
+                    "toolVersion": "1.0"
+                },
+                {
+                    "keyCode": "CP-230012",
+                    "operationSetCode": "SigntoolVerify",
+                    "parameters": [ ],
+                    "toolName": "sign",
+                    "toolVersion": "1.0"
+                }
+            ]
+          SessionTimeout: 20
+      - pwsh: |
+          $env:NewDir = '$(Pipeline.Workspace)\sign-summary'
+          New-Item -Path $env:NewDir -ItemType Directory -Force
+          Get-ChildItem '$(Pipeline.Workspace)\unzipped\*.md' | Foreach-Object { Move-Item -Path $_.FullName -Destination $env:NewDir\$($_.Name) }
+      - publish: $(Pipeline.Workspace)\sign-summary
+        artifact: sign-summary
+      - publish: $(Pipeline.Workspace)\unzipped
+        artifact: signed-executable
+
+- stage: zip
+  dependsOn: [scan, sign]
```

Comment:
no action here, just to mention I like the fact you parralelized the scan and sign operations, which can both take a while